python/lighthouse/workload.py

```diff
@@ -0,0 +1,77 @@
+"""
+Abstract base class for workloads.
+
+Defines the expected interface for generic workload execution methods.
+"""
+from mlir import ir
+from abc import ABC, abstractmethod
+from contextlib import contextmanager
+from typing import Optional
+
+
+class Workload(ABC):
+    """
+    Abstract base class for workloads.
+
+    A workload is defined by a fixed payload function and problem size.
+    Different realizations of the workload can be obtained by altering the
+    lowering schedule.
+
+    The MLIR payload function should take input arrays as memrefs and return
+    nothing.
+    """
+    payload_function_name: str = "payload"
+
+    @abstractmethod
+    def requirements(self) -> list[str]:
```

Comment:
How about each `Workload` being associated with a `Target` and each `Target` knows how obtain a proper `ExecutionEngine`?

Reply:
The default implementation of method that returns the associated `Target` could even look for `DLTI` on the payload and/or schedule IR, c.f. https://github.com/llvm/llvm-project/pull/145899/files#diff-3a6176958ee72a0dc93d8cca802e4d7d35a813c24cb24bede8d3bca7a8ad75caR55-R57, and based on that automagically try to instantiate the appropriate Lighthouse `Target`.

Reply:
Agreed -- we can leave this as is for now. Will leave this "unresolved" just as a reminder (at least for myself) for when we come to address representing targets in Lighthouse.